packages/angular/cli/src/commands/generate/cli.ts

```diff
@@ -162,4 +160,45 @@ export class GenerateCommandModule
 
     return `${commandName}${positionalArgs ? ' ' + positionalArgs : ''}`;
   }
+
+  private async getSchematicsToRegister(): Promise<
+    [schematicName: string, collectionName: string][]
```

Comment:
Nit: It might be a cleaner abstraction to have this return a `Map<schematicName: string, collectionName: string>` and then do the `.entires().sort()` in the caller. Up to you where we should enforce ordering.

Reply:
Same as https://github.com/angular/angular-cli/pull/22860#discussion_r829879240